### DISCUSSION TITLE
Customizing Widget Position in Theia's Left Panel

### DISCUSSION THREAD

**Discussion by Ashishh0007** [eclipse-theia/theia#14155] [Q&A]
I’ve created a custom widget in Theia and added it to the left panel’s default active bar. To set its position, I used the following code in the application-layout file:

app.shell.leftPanelHandler.addWidget(this.webServicesWidget, { rank: 9 });

However, the widget still appears in the second position from the top, instead of where I expected it to be.

Could someone guide me on how to customize the positions of both Theia’s default widgets and manually created ones? Is there a way to control their ordering in the left panel? Any suggestions would be greatly appreciated!

**Comment by Ashishh0007** [comment]
@msujew can you please help or suggest on this? It will really be a great help for us.

**Comment by msujew** [comment]
Hey @Ashishh0007,

As far as I'm aware, using the `rank` field should yield the resired effect. It should just be a lower value than it is defined here:

https://github.com/eclipse-theia/theia/blob/ac0c246fbcd439c259b4778e41199708d4354899/packages/navigator/src/browser/navigator-contribution.ts#L173

Which definitely is the case in your snippet. I'll take a look at this next week to see whether I'm able to reproduce this issue.

**Reply by NikshipthYathindra** [reply]
hi @msujew  i'm also working on the same and unable to place my custom widget after the explorer widget.

app.shell.leftPanelHandler.addWidget(this.CustomWidget, { rank: 110 });

currently i'm using theia 1.49.0 version. could please check once

**Reply by sowjanyagunupuru** [reply]
Hello @msujew 
I too working on changing the positions of the widgets visible on left panel 
Following the above references and snippets , I also tried with changing rank for the widget.

app.shell.leftPanelHandler.addWidget(this.fileSearchBarWidget, { rank: 10 });

But result is not satisfying requirement i.e the positions of the widgets are not changing.

Can u kindly suggest us the way to solve this issue to change these positions of the widgets
